### PR TITLE
cookbook modernization

### PR DESCRIPTION
**Pull request by lamont-granquist**
- travis
- cookstyle
- fix bats tests
- expand distro testing
- berkshelf
- etc

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -17,48 +17,45 @@
...
-  execute "Update locale" do
+  execute 'Update locale' do
     command "update-locale LANG=#{lang} LC_ALL=#{lc_all}"
-    not_if { Locale.up_to_date?("/etc/default/locale", lang, lc_all) }
+    not_if { Locale.up_to_date?('/etc/default/locale', lang, lc_all) }
   end
-
-when 'redhat', 'centos', 'scientific', 'amazon'
-  locale_file_path = "/etc/sysconfig/i18n"
+elsif File.exist?('/usr/bin/localectl')
```
Comment (tas50) [inline recipes/default.rb]: Is this just so we can continue to support older Chef releases that don't have the init system ohai attribute?

Comment (lamont-granquist) [inline recipes/default.rb]: I'd prefer to move to putting chef-sugar's `systemd?` helper into Chef::DSL::Universal.

Comment (tas50) [inline recipes/default.rb]: Sounds good to me. We'd have the same backwards compat problems though

Comment (lamont-granquist) [inline recipes/default.rb]: not if we leverage chef-sugar the same way as we leverage compat_resource (and or migrate the helper methods to compat_resource)

**Pull request merged**